guide/src/migration.md

```diff
@@ -0,0 +1,183 @@
+# Appendix B: Migration Guides for major version changes
```

Comment:
Technically `0.8` -> `0.9` is a minor version, so I don't think we need to be so specific here.
```suggestion
# Appendix B: Migrating from older PyO3 versions
```